htdocs/core/class/commondocgenerator.class.php

```diff
@@ -588,6 +588,21 @@ public function get_substitutionarray_lines($line, $outputlangs)
 		$line->fetch_optionals();
 
 		$resarray = $this->fill_substitutionarray_with_extrafields($line, $resarray, $extrafields, $array_key, $outputlangs);
+        
```

Comment:
If we call get_substitutionarray_lines and $line is ProposalLine or FactureLine for example, we also make a select on product_fournisseur_price_extrafields. We should do it only in some cases.